internal/monitors/statsd/monitor.go

```diff
@@ -0,0 +1,158 @@
+	Dimensions []string `yaml:"dimensions"`
+}
+
+// Config for this monitor
+type Config struct {
+	config.MonitorConfig `yaml:",inline" acceptsEndpoints:"false" singleInstance:"false"`
+	// The host/address on which to bind the UDP listener that accepts statsd
+	// datagrams
+	ListenAddress string `yaml:"listenAddress" default:"localhost"`
+	// The port on which to listen for statsd messages
+	ListenPort uint16 `yaml:"listenPort" default:"8125"`
+	// The protocol for statsd communications
+	ListenTCP    bool             `yaml:"listenTCP" default:"false"`
+	MetricPrefix string           `yaml:"metricPrefix"`
+	Converters   []converterInput `yaml:"converters"`
+}
+
+// Monitor that accepts and forwards trace spans
+type Monitor struct {
+	Output types.Output
+	cancel context.CancelFunc
+	conf   *Config
+}
+
+// Configure the monitor and kick off volume metric syncing
+func (m *Monitor) Configure(conf *Config) error {
+	var ctx context.Context
+	ctx, m.cancel = context.WithCancel(context.Background())
+
+	m.conf = conf
+
+	var converters []*converter
+	for _, ci := range conf.Converters {
+		converters = append(converters, initConverter(&converterInput{
+			Pattern:    ci.Pattern,
+			MetricName: ci.MetricName,
+			Dimensions: ci.Dimensions,
+		}))
+	}
+
+	listener := &statsDListener{
+		ipAddr:     conf.ListenAddress,
+		port:       conf.ListenPort,
+		tcp:        conf.ListenTCP,
+		prefix:     conf.MetricPrefix,
+		converters: converters,
+	}
+
+	err := listener.Listen()
+	if err != nil {
+		return err
+	}
+
+	go listener.Read()
+
+	utils.RunOnInterval(ctx, func() {
+		metrics := listener.FetchMetrics()
+		dps := convertMetricsToDatapoints(aggregateMetrics(metrics))
+
+		m.sendDatapoints(dps)
+	}, time.Duration(conf.IntervalSeconds)*time.Second)
+
+	return nil
+}
+
+func (m *Monitor) sendDatapoints(dps []*datapoint.Datapoint) {
+	for i := range dps {
+		m.Output.SendDatapoint(dps[i])
+	}
+}
+
+// Shutdown stops the forwarder and correlation MTSs
+func (m *Monitor) Shutdown() {
+	if m.cancel != nil {
+		m.cancel()
+	}
+}
+
+func aggregateMetrics(metrics []*statsDMetric) map[string]*statsDMetric {
+	metricsMap := make(map[string]*statsDMetric)
+
+	for _, metric := range metrics {
+		if _, exists := metricsMap[metric.metricName]; exists {
+			switch metricTypeMap[metric.metricType] {
+			case "gauge":
+				// Drop older metric by overwriting
+				metricsMap[metric.metricName] = metric
+			case "counter":
+				// Add up
+				metricsMap[metric.metricName].value = metricsMap[metric.metricName].value + metric.value
+			}
+		} else {
+			metricsMap[metric.metricName] = metric
+		}
+	}
+
+	return metricsMap
+}
+
+func convertMetricsToDatapoints(metrics map[string]*statsDMetric) []*datapoint.Datapoint {
+	var dps []*datapoint.Datapoint
+
+	for _, metric := range metrics {
+		var dp *datapoint.Datapoint
+
+		// StatsD Metric Types https://github.com/statsd/statsd/blob/master/docs/metric_types.md
+		switch metricTypeMap[metric.metricType] {
+		case "gauge":
```

Comment:
Use these const enums from golib: https://github.com/signalfx/golib/blob/master/datapoint/datapoint.go#L22.  Note that `Count` is a counter and `Counter` is a cumulative counter (it is kind of confusing).